File: Tienda_Pyton/Cliente.py
class Cliente:
    CANTIDAD_CLIENTES = 0

    def __init__(self, id=None, nombre=None, direccion=None):
        self.id = id
        self.nombre = nombre
        self.direccion = direccion
        self.tiene_compras = False
        if id is not None and nombre is not None and direccion is not None:
            Cliente.CANTIDAD_CLIENTES += 1

    def obtener_informacion(self):
        return f"ID: {self.id} nombre: {self.nombre} y direccion {self.direccion}"

    def set_tiene_compras(self, tiene_compras):
        self.tiene_compras = tiene_compras

    def get_nombre(self):
        return self.nombre
    def get_tiene_compras(self):
        return self.tiene_compras

File: Tienda_JAVA/src/javaapplication45/Cliente.java

package javaapplication45;
import java.util.*;
public class Cliente {
    private static int CANTIDAD_CLIENTES = 0;
    private int id;
    private String nombre;
    private String direccion;
    private boolean tieneCompras;

    public Cliente() {
    }
    
    public Cliente( int id, String nombre, String direccion) {
        this.nombre = nombre;
        this.direccion = direccion;
        this.id = id;
        this.tieneCompras=false;
        CANTIDAD_CLIENTES++;
    }

    public int getId() {
        return id;
    }

    public  int getCANTIDADCLIENTES() {
        return CANTIDAD_CLIENTES;
    }
    
    
    public String ObtenerInformacion(){
        return String.format("ID: %d nombre: %s y direccion %s", this.id, this.nombre, this.direccion);
    }

    public String getNombre() {
        return nombre;
    }

    public void setTieneCompras(boolean tieneCompras) {
        this.tieneCompras = tieneCompras;
    }

    public boolean isTieneCompras() {
        return tieneCompras;
    }
    
}
